index.bs

```diff
@@ -116,12 +116,15 @@ Framework {#framework}
     // ...
     // More things from https://github.com/cure53/DOMPurify/blob/master/src/purify.js#L224
   };
+  typedef [StringContext=TrustedHTML] DOMString HTMLString;
+  typedef (DOMString or HTMLString or DocumentFragment or Document) SanitizerInput;
```

Comment:
`DOMString or HTMLString or ...` I suspect this isn't going to work. The extended attribute doesn't really create a new type, and the result of this is that we now have a union with the same type twice, and the bindings won't know how to convert a given value into which of the two `DOMString`s.

(To be honest, I'm not super familiar with WebIDL and the bindings compiler, so maybe my understanding here is flawed...)

So: From the WebIDL (and bindings') perspective, we have three things:
- ` TrustedHTML` (this is the "trusted" type)
- `DOMString` (just a regular string)
- `[ScriptContext=String] DOMString`  (still a regular string)

That last thing is still a regular DOMString, but we're telling the bindings that this value will be used for something HTML-ish. Trusted Types then comes in and generates an additional check & conversion logic when it sees these. But this doesn't really change the underlying type; it's more a hint to the bindings compiler to add additional behaviour to the bindings.

So: If we want all methods to have the same input type(s), then we should probably have a typedef over `DOMString or TrustedHTML or ..` In that case, the bindings won't add special behaviour (and couldn't, because the behaviour would slightly differ depending on the method), and we'd add the checks manually. (No big deal, though.)

Or, we want to rely on the bindings to generate the checks. In that case, we can't quite preserve the symmetry between the methods and need to tell the bindings (via the extended attribute) that for two of these methods the input string will be used in HTML-ish context, while for the string-to-string method it won't.

Reply:
Removed _DOMString_. Hopefully this now makes sense 😊